### PR TITLE
Simplify check for tight-bbox finiteness.

### PR DESCRIPTION
**Pull request by anntzer**
`0 < bbox.width < np.inf` is as explicit as `bbox.width != 0 and
np.isfinite(bbox.width)`, shorter, and faster (not that it is a
bottleneck, though).  (It works because bboxes are oriented to have
nonnegative widths and heights.)

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/axis.py [hunk 1]
```diff
@@ -1152,20 +1152,15 @@ def get_tightbbox(self, renderer):
         self.offsetText.set_text(self.major.formatter.get_offset())
 
         bb = []
-
         for a in [self.label, self.offsetText]:
-            bbox = a.get_window_extent(renderer)
-            if (np.isfinite(bbox.width) and np.isfinite(bbox.height) and
```
Comment (anntzer) [inline lib/matplotlib/axis.py]: this finiteness check is redundant with the one below

File: lib/matplotlib/axes/_base.py [hunk 2]
```diff
@@ -4335,8 +4335,7 @@ def get_tightbbox(self, renderer, call_axes_locator=True,
         for a in bbox_artists:
             bbox = a.get_tightbbox(renderer)
             if (bbox is not None and
-                    (bbox.width != 0 or bbox.height != 0) and
-                    np.isfinite(bbox.width) and np.isfinite(bbox.height)):
+                    0 < bbox.width < np.inf and 0 < bbox.height < np.inf):
```
Comment (timhoffm) [inline lib/matplotlib/axes/_base.py]: Optional, but might be easier to read when wrapping the height check to a separate line.

File: lib/matplotlib/axis.py [hunk 3]
```diff
@@ -1152,20 +1152,15 @@ def get_tightbbox(self, renderer):
         self.offsetText.set_text(self.major.formatter.get_offset())
 
         bb = []
```
Comment (timhoffm) [inline lib/matplotlib/axis.py]: While we're at it, could be renamed to `bboxes`. Optional.

**Comment by anntzer** [comment]
both comments handled

**Comment by jklymak** [comment]
Is it worth doing `bboxes = cbook.valid_bboxes(bboxes)`?

**Comment by tacaswell** [comment]
The values can be `np.nan` and 

```
In [33]: np.nan < np.inf
Out[33]: False

```

See https://github.com/matplotlib/matplotlib/pull/13426

**Comment by anntzer** [comment]
But... that's the point here: these are places that check `width != 0 and isfinite(width)` so return False for nan, and `0 < width < inf` also return False for nan.

**Comment by tacaswell** [comment]
:sheep: Sorry, was sleepy and got confused about the polarity we were looking at here.

**Comment by tacaswell** [comment]
Thanks @anntzer

**Pull request merged**